### PR TITLE
Feature/event bridge

### PR DESCRIPTION
**Pull request by fatihaydilek**

### CODE REVIEW

File: src/Constants.ts [hunk 1]
```diff
@@ -229,10 +229,16 @@ export const ClassNames = {
...
     ELASTICSEARCH: 'ELASTICSEARCH',
     MYSQL: 'MYSQL',
+    EVENTBRIDGE: 'AWS-EventBridge',
 };
 
 export const AWS_SERVICE_REQUEST = 'AWSServiceRequest';
 
+export const AwsEventBridgeTags = {
+    SERVICE_REQUEST: 'AWSEventBridgeRequest',
+    EVENT_BUS_NAME: 'event.bus.name',
```
Comment (serkan-ozal) [inline src/Constants.ts]: I think, the `EVENT_BUS_NAME` tag name should be `aws.eventbridge.eventbus.name`

File: src/Constants.ts [hunk 2]
```diff
@@ -517,6 +523,7 @@ export const SpanTags = {
     TRIGGER_OPERATION_NAMES: 'trigger.operationNames',
     TOPOLOGY_VERTEX: 'topology.vertex',
     TRACE_LINKS: 'trace.links',
+    SPAN_RESOURCES: 'span.resources',
```
Comment (serkan-ozal) [inline src/Constants.ts]: I think, the constant name should be `RESOURCE_NAMES` and tag name should be `resource.names` accordingly

File: src/plugins/utils/LambdaEventUtils.ts [hunk 3]
```diff
@@ -214,6 +217,25 @@ class LambdaEventUtils {
         return className;
     }
 
+    static injectTriggerTagsForEventBridge(span: ThundraSpan, originalEvent: any): String {
+        const domainName = DomainNames.MESSAGING;
+        const className = ClassNames.EVENTBRIDGE;
+        const traceLinks: any[] = [];
+        const eventDetails: Set<string> = new Set<string>();
+
+        console.log('eventbridge inject ', JSON.stringify(originalEvent));
```
Comment (hamitb) [inline src/plugins/utils/LambdaEventUtils.ts]: 👆

Comment (serkan-ozal) [inline src/plugins/utils/LambdaEventUtils.ts]: Remove log statement

File: src/plugins/support/InvocationTraceSupport.ts [hunk 4]
```diff
@@ -23,12 +23,26 @@ class InvocationTraceSupport {
                             filter((span: ThundraSpan) => span.spanContext.spanId !== rootSpanId);
 
             for (const span of spans) {
-                const resourceId = InvocationTraceSupport.generateResourceIdFromSpan(span);
-                if (resourceId) {
-                    const resource = resourcesMap.get(resourceId);
-                    const newResource = new Resource();
-                    newResource.init(span);
-                    resource ? resource.merge(newResource) : resourcesMap.set(resourceId, newResource);
+                const entries = span.getTag(SpanTags.SPAN_RESOURCES);
```
Comment (serkan-ozal) [inline src/plugins/support/InvocationTraceSupport.ts]: I think, this should be renamed as `resourceNames`

File: src/plugins/support/InvocationTraceSupport.ts [hunk 5]
```diff
@@ -23,12 +23,26 @@ class InvocationTraceSupport {
...
+                if (entries) {
+                    for (const entry of entries) {
```
Comment (serkan-ozal) [inline src/plugins/support/InvocationTraceSupport.ts]: I think, this should be renamed as `for (const resourceName of resourceNames)`

File: src/plugins/support/InvocationTraceSupport.ts [hunk 6]
```diff
@@ -23,12 +23,26 @@ class InvocationTraceSupport {
...
+                        const entryId = InvocationTraceSupport.generateResourceIdFromSpan(span, entry);
```
Comment (serkan-ozal) [inline src/plugins/support/InvocationTraceSupport.ts]: I think, `entryId` should be renamed as `resourceId`

File: src/plugins/support/InvocationTraceSupport.ts [hunk 7]
```diff
@@ -39,9 +53,12 @@ class InvocationTraceSupport {
         }
     }
 
-    static generateResourceIdFromSpan(span: ThundraSpan): string {
+    static generateResourceIdFromSpan(span: ThundraSpan, entryId?: string): string {
```
Comment (serkan-ozal) [inline src/plugins/support/InvocationTraceSupport.ts]: I think, `entryId` should be renamed as `resourceName`

File: src/plugins/support/InvocationTraceSupport.ts [hunk 8]
```diff
@@ -39,9 +53,12 @@ class InvocationTraceSupport {
...
         if (span.className && span.operationName) {
             let id = `${span.className.toUpperCase()}\$${span.operationName}`;
+            if (entryId) {
```
Comment (serkan-ozal) [inline src/plugins/support/InvocationTraceSupport.ts]: If `resourceName` (new name of `entryId` parameter) is passed, `operationName` should not be used in the generated entry id.
```
if (!resourceName) {
    resourceName = span.operationName;
}
let id = `${span.className.toUpperCase()}\$${resourceName}`;

**Pull request merged**